src/streamlit_app.py

```diff
@@ -36,97 +38,229 @@
 
-The date today is {today}
+The date is {today}
 
-# Your Task
+## Instructions
 
 You will be provided with a schema. A user will ask a question about the schema or about data using that schema.
 Determine if this a question about schema or data.
- * If about schema, answer in markdown.
- * If about data, you will create a function call to `HONEYDEW` to provide data for the user question.
-The function looks like
-```
-CALL HONEYDEW({{
-  "metrics": [ "table1.met1", "table2.met2", ... ],
-  "group_by": ["table1.attr1", "table2.attr2", ... ],
-  "filters": ["table2.attr3 = ''''string_val''''", "table2.attr3 >= number_val", ... ]
-}})
-```
+- Carefully read the user's question to comprehend the request.
+- If about schema, answer in markdown.
+- If about data, you will return a JSON query object as defined and exampled below
+  - Identify key terms, entities, metrics, attributes, aggregations, and filters.
+  - Determine the intent of the question and what the user is seeking to find out.
 
-## Example - data
-For example, user asks for revenue in brooklyn by customer type and month since 2023.
-A response will be:
-```
-CALL HONEYDEW({{
-  "metrics": [ "orders.revenue" ],
-  "group_by": ["customer.type", "date.month" ],
-  "filters": ["location.name = ''''Brooklyn''''", "YEAR(date.date) >= 2023" ]
-}})
-```
+## Response Rules
 
-## Example - schema
-For example, user asks for which metrics there are about customers
-A response will be a list of metrics relevant to customers in the schema
+- errors
+   - Go over the matching metadata, look carefully and perform the following validations
+   - **Term Not Found**: Note any terms that cannot be matched to the schema.
+   - **Ambiguous Term**: If a term matches multiple schema items equally, note all possible matches.
+   - **Aggregation Mismatch**: If the requested aggregation doesn't match the metric's defined aggregation (and is not equivalent), report an aggregation mismatch error.
+   - **Metrics Validation**:
+     - **Ensure only metrics are in the `metrics` array. Do not include attributes in the `metrics` array.**
+     - **Ensure the aggregation matches the metric's defined aggregation in the schema, considering equivalent aggregations.**
+
+- if error occured
+  - provide messages according to the error handling guidelines.
+  - Output text if there something missing or the question is about the schema.
+
+- if data question
+  - Apply filters according to the rules specified.
+  - only attributes can be referenced in the "group_by" array
+  - only metrics can be referenced in the metrics array
+  - present the rewritten question highlighting the attributes, metrics & filters you've successfully matched to the user query
+  - present the JSON query.
+
+### Response Structure
+
+```json
+{{
+  "group_by": ["entity.attribute_name"],
+  "metrics": ["entity.metric_name"],
```

Comment:
metrics are optional for data retriveal questions

Reply:
Fixed